src/nxt_socket.c

```diff
@@ -285,6 +285,20 @@ nxt_socket_close(nxt_task_t *task, nxt_socket_t s)
     nxt_err_t   err;
     nxt_uint_t  level;
 
+#if (NXT_HAVE_UNIX_DOMAIN)
+    socklen_t           len;
+    struct sockaddr_un  sa = {0};
+
+    len = sizeof(struct sockaddr_un) - 1;
+    if ((getsockname(s, (struct sockaddr *)&sa, &len) == 0)
+        && (sa.sun_family == AF_UNIX))
+    {
+        ((char *) &sa)[len] = '\0';
```

Comment:
The opening **{** should be on the preceding line I believe.

Also, won't sa.sun_len be null-terminated from the 0 initialisation of sa? Or do you fear some compilers won't get it right?

I take it the -1 on the length was to do with null-terminating sa.sun_path?

Reply:
From the coding style <https://nginx.org/en/docs/dev/development_guide.html#code_style>:

> Conditionals and Loops
> 
> The “if” keyword is separated from the condition by one space. Opening brace is located on the same line, or on a dedicated line if the condition takes several lines.

Reply:
Re: NUL-termination:  Oh yes, that's a glitch.  I first terminated it by one method, then changed it, and forgot to remove the old code.

My concern with `{0}` is that I'm not sure if I can trust the kernel not writing extra bytes with garbage, as I've never had to use getsockname(2) before and am not used to it.  I guess that letting garbage into user-space data could be a security risk and probably the kernel doesn't do that, but wanted to be sure.  So, can you confirm that I can just remove the `... = '\0';` assignment?

The behavior of `getsockname(2)` regarding the string should be documented in the manual page.  I read it several times and it says nothing about it.  Could you please send a patch if you know how it works?

Reply:
RE if. Ah, I'll need to change some of my if's that have been combined and now fall into that category...

Actually, I didn't think to look at the nginx coding style. The unit CONTRIBUTING.md does say this

> please read the NGINX Unit code guidelines to know more about coding conventions

Is that actually referring to the nginx coding style document?

Regarding

> My concern with {0} is that I'm not sure if I can trust the kernel not writing extra bytes with garbage, as I've never had to use getsockname(2) before and am not used to it.

I would certainly hope it doesn't. On Linux I think everything is good, from reading the unix(7) man page, you get back a null-terminated sun_path.

However the BUGS section has a caveat to that in that if you supplied a pathname of 108 characters with no nul byte, the kernel will add one which will exceed sizeof(sun_path) and depending on how you set addrlen you may not get back the nul byte. More interestingly is what it says about other implementations where they don't require a null-terminated string when binding a socket in which case it basically suggests doing what you've done ;).

Hmm, I wonder if we do any checking on the path length when creating the socket...

Reply:
Re: unix(7):  Hmm, it seems (although the text isn't entirely clear, that's what I infer) that systems respect the NUL byte if you pass it, so as long as we guarantee that we create sockets with a terminating NUL byte, I don't need to care at all about it.

Reply:
Yes, that is my understanding also. Linux will add one if it's missing, some others won't, but if you put one, then it _should_ stay there.